scalafix-tests/output-dotty/src/main/scala/test/DottyAutoTuplingFunctionArgs.scala

```diff
@@ -0,0 +1,37 @@
+package test
+
+object DottyAutoTuplingFunctionArgs {
+  Some((1, 2)).map((a, b) => a + b)
+  List((1, 2, 3), (4, 5, 6)).map { (a, b, c) => println(a); a + b + c }
+  List((1, 2), (3, 4)).foreach { (a, b) => { val c = a + b; println(c) } }
+  Set((1, 2, 3)).map { (a, b, c) =>
+    println(a + b + c)
+  }
+  List((1, 1, 1, 1, 1), (2, 2, 2, 2, 2)).flatMap{
+    (_, _, _, _, n) => List(n)
+  }
+  val xs = List("3", "1", "0").zipWithIndex
+  xs.filter((w, i) => w == i.toString)
+
+  // should not rewrite the below
+  xs.map { case (w, i) if i > 1 => s"$w-$i" } // case with guards
+  List((Some(1), Some(1))).map { case (Some(a), Some(b)) => a + b } // extraction
+  xs.map {
+    case (a, b) => a + b
+    case _ => 0
+  } // multiple cases
+  xs.find { case (_, x @ b) => x - b == 0 } // binding
+  xs.map { case ("0", 0) => "zero" } // literals
+  List(("a", 1, false, 1.0), ("b", 2, true, 2.0)).exists{
+    case (_: String, _: Int, c: Boolean, _: Double) => c
+  } // typed parameters
+  class A
+  class B extends A
+  def foo(xs: List[(A, A)]) = xs.map {
+    case (x: B, y: A) => 1
+  } // typed parameters not matching tuple element types
+  def any(xs: List[Any]): List[Int] = xs.map { case (x, y) => 1 }
+  def anyIntPair(xs: List[(Int, Any)]): List[Int] = xs.map((x, y) => 1)
+  def anyAnyPair(xs: List[(Any, Any)]): List[Int] = xs.map { case (x, y) => 1 } // false negative
+  def tparam[T](xs: List[T]): List[Int] = xs.map { case (x, y) => 1 }
```

Comment:
there is one comment in line 16 which separates the scenarios that should be rewritten from the ones that should not. Maybe move the line 34 to somewhere before that comment?

Reply:
Good idea, moved these tests above that comment.